plugins/parsers/json/parser.go

```diff
@@ -195,10 +195,13 @@ func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	if p.query != "" {
 		result := gjson.GetBytes(buf, p.query)
 		buf = []byte(result.Raw)
-		if !result.IsArray() && !result.IsObject() {
-			err := fmt.Errorf("E! Query path must lead to a JSON object or array of objects, but lead to: %v", result.Type)
+		if !result.IsArray() && !result.IsObject() && result.Type != gjson.Null {
+			err := fmt.Errorf("E! Query path must lead to a JSON object, array of objects or null, but lead to: %v", result.Type)
 			return nil, err
 		}
+		if result.Type == gjson.Null {
+			return make([]telegraf.Metric, 0), nil
```

Comment:
I think we should be able to return nil here
```suggestion
			return nil, nil
```

Reply:
Updated, thanks!